### PR TITLE
Refactored Disabled component to use React Hooks

### PR DESCRIPTION
**Pull request by javidalkaruzi**
## Description
See #22890

Refactored the Disabled component to use React Hooks.

## How has this been tested?
Tested using existing unit tests.
Observed in Storybook that the components specified were disabled when rendered and automatically disabled again if I, the user, manipulate the DOM

## Types of changes
Code refactor

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [ ] My code follows the accessibility standards.
- [ ] My code has proper inline documentation.
- [ ] I've included developer documentation if appropriate.
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/disabled/index.js [hunk 1]
```diff
@@ -31,40 +36,11 @@ const DISABLED_ELIGIBLE_NODE_NAMES = [
 	'TEXTAREA',
 ];
 
-class Disabled extends Component {
-	constructor() {
-		super( ...arguments );
+function Disabled( { className, children, ...props } ) {
+	const node = useRef( null );
```
Comment (youknowriad) [inline packages/components/src/disabled/index.js:40]:
```suggestion
	const node = useRef();
```

I don't think it makes any difference but was wondering why setting it explicitly here.

**Review by youknowriad — APPROVED** [review]

**Comment by youknowriad** [comment]
Great refactoring, thanks. Can you rebase the PR to fix the checks?

File: packages/components/src/disabled/index.js (hunk 1 above)
Comment (javidalkaruzi) [reply packages/components/src/disabled/index.js]: I don't think it made any difference.

**Comment by youknowriad** [comment]
Looks like you messed up the branch while rebasing? (both PRs)

**Comment by javidalkaruzi** [comment]
> Looks like you messed up the branch while rebasing? (both PRs)

Yeah. I had rebased master into these branches.

**Pull request closed without merging**